yt/data_objects/tests/test_extract_regions.py

```diff
@@ -46,3 +48,16 @@ def test_cut_region():
         assert_equal(p2["density"].max() > 0.25, True)
         p2 = ds.proj("density", 2, data_source=cr, weight_field = "density")
         assert_equal(p2["density"].max() > 0.25, True)
+
+
+def test_region_and_particles():
```

Comment:
This function needs to be decorated with `@requires_module('scipy')`

Another way to do this would be to leave the brute-force algorithm there for cases when scipy isn't installed, in which case you should probably test both cases here.

Reply:
scipy is _not_ required as yt is shipped along with scipy's KDTree python-based implementation. Here we try to import scipy's cython-based version, but defaults to the python-based one.

Reply:
I've commented a bit too fast; actually the version shipped with yt also depends on scipy. I'm going to update the tests.

Reply:
So it looks like this test is still failing on the minimal install on travis without scipy, this time with an outright failure. Since you replaced the old brute-force search with the tree I guess this not working without scipy is expected, although I would expect it to raise an ImportError instead of giving an incorrect result.

Can we make it so it raises an ImportError and then decorate this test with `@requires_module('scipy')`? If not can you explain in a little more detail what happens when scipy isn't installed in this PR?

Reply:
In the case where `scipy` is not installed, it should fall back to the old brute-force search which *should* be equivalent to the KD-tree one. However, it seems that the old brute-force is not returning the same answers as the KD-tree one, so one of them must be incorrect